test/new-e2e/utils/e2e/client/docker.go

```diff
@@ -99,11 +113,34 @@ func (docker *Docker) ExecuteCommandWithErr(containerName string, commands ...st
 	return output, errOutput, err
 }
 
+// GetClient gets the [docker client].
+//
+// [docker client]: https://pkg.go.dev/github.com/docker/docker/client
 func (docker *Docker) GetClient() *client.Client {
 	return docker.client
 }
 
+// GetAgentContainerName gets the agent container name
 func (docker *Docker) GetAgentContainerName() string {
 	require.NotEmptyf(docker.t, docker.agentContainerName, "agent container not found")
 	return docker.agentContainerName
 }
+
+// GetAgentCommandRunner gets an agent that provides high level methods to run Agent commands.
+func (docker *Docker) GetAgentCommandRunner() *AgentCommandRunner {
+	require.NotNilf(docker.t, docker.agent, "there is no agent installed on this docker instance")
+	return docker.agent
+}
+
+var _ agentRawCommandRunner = (*dockerAgentCommand)(nil)
+
+// dockerAgentCommand is a wrapper to execute Agent commands on Docker.
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
What about calling this `dockerAgentRawCommandRunner` ?

Reply:
It would be removed when handling https://github.com/DataDog/datadog-agent/pull/18161#discussion_r1268279436.